thoth/python/source.py

```diff
@@ -405,3 +406,13 @@ def get_package_hashes(self, package_name: str, package_version: str, with_inclu
             result.append(doc)
 
         return result
+
+    def get_package_release_date(self, package_name: str, package_version: str,) -> datetime:
+        """Get time at which package was uploaded to package index."""
+        package_json = self._warehouse_get_api_package_info(package_name)
+        release = package_json["releases"][package_version]
+        artifact = next(x for x in release if x["python_version"] == "source")
+        if artifact is None:
+            raise ValueError(f"No source distribution found on {self.warehouse_api_url}")
```

Comment:
You can also add name and version of the package so the exception is self-descriptive and easily identified when encountered in the cluster.

Reply:
Also, you can use thoth-python exception `NotFound`, so anyone using thoth-python can handle library exceptions (respecting exception hierarchy).